pkg/mutation/match/match.go

```diff
@@ -22,7 +22,13 @@ const Wildcard = "*"
 // Match selects objects to apply mutations to.
 // +kubebuilder:object:generate=true
 type Match struct {
-	Kinds []Kinds `json:"kinds,omitempty"`
+	// Source determines if the mutator should be used to expand generator
+	// resources. Accepts `Generated`|`Original`|`All` (defaults to `Original`). A
+	// value of `Generated` will cause the mutator only be applied when expanding
+	// generator resources, while `Original` will cause the mutator to be applied
+	// resources going through the mutation webhook.
+	Source string  `json:"source,omitempty"`
```

Comment:
Here is where we should implement the logic for matching Source (and handling the case where source is undefined, which means match everything)